File: lib/domain/entities/family_member_entity.dart
import 'package:equatable/equatable.dart';

class FamilyMemberEntity extends Equatable {
  final String id;
  final String userId;
  final String name;
  final String relationship;
  final DateTime? dateOfBirth;
  final String? bloodType;
  final String? allergies;
  final String? medicalConditions;
  final String? emergencyContact;
  final String? notes;
  final DateTime createdAt;
  final DateTime updatedAt;

  const FamilyMemberEntity({
    required this.id,
    required this.userId,
    required this.name,
    required this.relationship,
    this.dateOfBirth,
    this.bloodType,
    this.allergies,
    this.medicalConditions,
    this.emergencyContact,
    this.notes,
    required this.createdAt,
    required this.updatedAt,
  });

  @override
  List<Object?> get props => [
        id,
        userId,
        name,
        relationship,
        dateOfBirth,
        bloodType,
        allergies,
        medicalConditions,
        emergencyContact,
        notes,
        createdAt,
        updatedAt,
      ];

  FamilyMemberEntity copyWith({
    String? id,
    String? userId,
    String? name,
    String? relationship,
    DateTime? dateOfBirth,
    String? bloodType,
    String? allergies,
    String? medicalConditions,
    String? emergencyContact,
    String? notes,
    DateTime? createdAt,
    DateTime? updatedAt,
  }) {
    return FamilyMemberEntity(
      id: id ?? this.id,
      userId: userId ?? this.userId,
      name: name ?? this.name,
      relationship: relationship ?? this.relationship,
      dateOfBirth: dateOfBirth ?? this.dateOfBirth,
      bloodType: bloodType ?? this.bloodType,
      allergies: allergies ?? this.allergies,
      medicalConditions: medicalConditions ?? this.medicalConditions,
      emergencyContact: emergencyContact ?? this.emergencyContact,
      notes: notes ?? this.notes,
      createdAt: createdAt ?? this.createdAt,
      updatedAt: updatedAt ?? this.updatedAt,
    );
  }
}

File: backend/src/family/entities/family-member.entity.ts
import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';

export enum Gender {
  MALE = 'male',
  FEMALE = 'female',
  OTHER = 'other',
}

export enum BloodGroup {
  A_POSITIVE = 'A+',
  A_NEGATIVE = 'A-',
  B_POSITIVE = 'B+',
  B_NEGATIVE = 'B-',
  O_POSITIVE = 'O+',
  O_NEGATIVE = 'O-',
  AB_POSITIVE = 'AB+',
  AB_NEGATIVE = 'AB-',
}

export class FamilyMember {
  @ApiProperty()
  id: string;

  @ApiProperty()
  userId: string;

  @ApiProperty({ example: 'Jane Doe' })
  name: string;

  @ApiProperty({ example: 'Spouse' })
  relation: string;

  @ApiPropertyOptional()
  dateOfBirth?: Date;

  @ApiPropertyOptional({ enum: BloodGroup })
  bloodGroup?: BloodGroup;

  @ApiPropertyOptional({ enum: Gender })
  gender?: Gender;

  @ApiPropertyOptional({ example: 165 })
  height?: number;

  @ApiPropertyOptional({ example: 60 })
  weight?: number;

  @ApiPropertyOptional({ example: ['Diabetes'] })
  medicalConditions?: string[];

  @ApiPropertyOptional({ example: ['Peanuts'] })
  allergies?: string[];

  @ApiPropertyOptional()
  notes?: string;

  @ApiProperty()
  createdAt: Date;

  @ApiProperty()
  updatedAt: Date;
}
